feature_engine/datetime/datetime_subtraction.py

```diff
@@ -0,0 +1,272 @@
+        ):
+            raise ValueError(
+                "reference must be a list of strings or integers. "
+                f"Got {reference} instead."
+            )
+
+        if not isinstance(drop_original, bool):
+            raise ValueError(
+                "drop_original takes only booleans True or False. "
+                f"Got {drop_original} instead."
+            )
+
+        if missing_values not in ["raise", "ignore"]:
+            raise ValueError(
+                "missing_values takes only values 'raise' or 'ignore'. "
+                f"Got {missing_values} instead."
+            )
+
+        valid_output_units = {'D', 'Y', 'M', 'W', 'h', 'm', 's', 'ms', 'us', 'μs', 'ns',
+                              'ps', 'fs', 'as'}
+
+        if output_unit not in valid_output_units:
+            raise ValueError(f"output_unit accepts the following values: "
+                             f"{valid_output_units}")
+
+        self.variables = variables
+        self.reference = reference
+        self.drop_original = drop_original
+        self.missing_values = missing_values
+        self.output_unit = output_unit
+
+    def fit(self, X: pd.DataFrame, y: Optional[pd.Series] = None):
+        """
+        This transformer does not learn any parameter.
+
+        Parameters
+        ----------
+        X: pandas dataframe of shape = [n_samples, n_features]
+            The training input samples. Can be the entire dataframe, not just the
+            variables to transform.
+
+        y: pandas Series, or np.array. Default=None.
+            It is not needed in this transformer. You can pass y or None.
+        """
+        # Common checks and attributes
+        X = check_X(X)
+
+        # check variables are datetime
+        self.reference = _find_or_check_datetime_variables(X, self.reference)
+        self.variables = _find_or_check_datetime_variables(X, self.variables)
+
+        # check if dataset contains na
+        if self.missing_values == "raise":
+            _check_contains_na(X, self.variables)
+            _check_contains_inf(X, self.variables)
+
+        # save input features
+        self.feature_names_in_ = X.columns.tolist()
+
+        # save train set shape
+        self.n_features_in_ = X.shape[1]
+
+        return self
+
+    def transform(self, X: pd.DataFrame) -> pd.DataFrame:
+        """
+        Add new features.
+
+        Parameters
+        ----------
+        X: pandas dataframe of shape = [n_samples, n_features]
+            The data to transform.
+
+        Returns
+        -------
+        X_new: Pandas dataframe
+            The input dataframe plus the new variables.
+        """
+
+        # Check method fit has been called
+        check_is_fitted(self)
+
+        # check that input is a dataframe
+        X = check_X(X)
+
+        # Check if input data contains same number of columns as dataframe used to fit.
+        _check_X_matches_training_df(X, self.n_features_in_)
+
```

Comment:
we need to add the check for missing data in the transform method as well if missing_only is raise.